### PR TITLE
Lots of logic for the build struct

### PR DESCRIPTION
**Pull request by gordallott**
now Build should be able to run builds which is nice, tests show it
works

not everything is tested, i ran out of steam a bit, the important bit is

also added a Config() to get the config from a build

### CODE REVIEW

File: core/build.go [hunk 1]
```diff
@@ -46,14 +94,275 @@ func (b *build) hasFinished() bool {
...
+}
 
 // Stop will stop the given build, it will error with ErrAlreadyStopped if the build has finished
-func (b *build) Stop() error { return nil }
+func (b *build) Stop() error {
+	if b == nil {
+		return errors.New("b is nil")
+	}
+
+	b.m.RLock()
```
Comment (njpatel) [inline core/build.go]: Feels like just doing a defer b.m.Unlock here would make the code below a bit cleaner

File: core/exists.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package core
+
+import (
+	"os"
+	"path/filepath"
+)
+
+// Exists returns whether the given file or directory exists or not
```
Comment (njpatel) [inline core/exists.go]: So sad, so so so sad

**Comment by njpatel** [comment]
Looks good, just that comment above

File: core/build.go (hunk 1 above)
Comment (gordallott) [inline core/build.go]: it would, this didn't used to be just this lock

**Pull request merged**